File: js/day59.js
// Check if string contains 8 or 10 digits (even if there's dash, space or + parenthesis) (02) 8-555-6677

const str = "(02) 8-555-6677";

const digitsOnly = str.replace(/\D/g, '');
const isValid = digitsOnly.length === 8 || digitsOnly.length === 10;
console.log(isValid);

File: python/day59.py
import re

# Check if string contains 8 or 10 digits (even if there's dash, space or + parenthesis) (02) 8-555-6677

text = "(02) 8-555-6677"

digits_only = re.sub(r'\D', '', text)
is_valid = len(digits_only) in (8, 10)
print(is_valid)